staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresource/tableconvertor/tableconvertor.go

```diff
@@ -107,17 +107,26 @@ func (c *convertor) ConvertToTable(ctx context.Context, obj runtime.Object, tabl
 				continue
 			}
 
-			// as we only support simple JSON path, we can assume to have only one result (or none, filtered out above)
-			value := results[0][0].Interface()
 			if customHeaders[i].Type == "string" {
-				if err := column.PrintResults(buf, []reflect.Value{reflect.ValueOf(value)}); err == nil {
+				values := []reflect.Value{}
+				for i, v := range results[0] {
+					if i >= 3 {
+						more := len(results[0]) - i
+						values = append(values, reflect.ValueOf(fmt.Sprintf("and %d more", more)))
```

Comment:
This always strikes me as problematic.  Why 3?  I think i'd probably prefer to just print all of them and it's up to the client to make it look good.  For kubectl we have a couple of options, but even the naive view isn't bad.

Reply:
Also, if you have that many items, i think that's an argument for not returning them as a string but instead an array which gives the client more options.

Reply:
Does jsonpath have an effective slice selector?  If so, I'd say we could also punt this to jsonpath.

Reply:
Yeah I wasn't sure what to do here. For Gateway API we have a real concern that we'd end up with a list of 100 matching entries in some cases, so I'm trying to avoid that. This took an approach similar to how we print Endpoints and EndpointSlices. With that said, if there's a way to accomplish that in kubectl instead, I'm very open to that. Any tips for how I could implement this elsewhere?

Reply:
in general server side print is trying give users the right experience with a simple model clients can handle.  CRD support is trying to offer some bounded flexibility for authors, but within what the server can handle.

Do we correctly support the json path [0:3] syntax?  A CRD author could decide to show the first three regardless.  Alternatively, we could define a new format accepted by the client.  For now, i’d say that displaying the array or string as is is “safe” and we can add complexity in a follow up (vs blocking here).

Reply:
OK, one more update, I was able to get this to treat multiple matches in the same way it would treat a single matching slice, but it's quite hacky. Very open to any ideas you might have to make it less messy.